File: 2015/02/python/main.py
import re

# --- Constants --
INPUT_FILE = "../input/input.txt"
REGEX = r"(\d+)x(\d+)x(\d+)"

# --- Variables ---
total = 0
ribbon = 0

# --- Read and Parse the input file --
file = open(INPUT_FILE, "r")
text = file.read().strip()
for cap in re.findall(REGEX, text):
    l = int(cap[0])
    w = int(cap[1])
    h = int(cap[2])
    a = l * w
    b = w * h
    c = l * h
    m = min(a, b, c)
    total += 2 * (a + b + c) + m

    m = max(l, w, h)
    ribbon += 2 * (l + w + h - m) + l * w * h

# --- Results ---
print(f"1. Total square feet of wrapping paper: {total:,d}")
print(f"2. Total feet of ribbon: {ribbon:,d}")

File: 2015/02/go/main.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"regexp"
	"strconv"
	"strings"
)

// --- Constants ---
const FILENAME = "../input/input.txt"
const REGEX = `(\d+)x(\d+)x(\d+)`

func main() {
	// --- Helper functions ---
	parseInt := func(s string) int {
		n, err := strconv.Atoi(s)
		if err != nil {
			log.Panic(err)
		}
		return n
	}

	// --- Variables ---
	re := regexp.MustCompile(REGEX)
	total := 0
	ribbon := 0

	// --- Read input file ---
	file, err := os.Open(FILENAME)
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	// --- Parse each line ---
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := strings.TrimSpace(scanner.Text())
		m := re.FindStringSubmatch(line)
		l := parseInt(m[1])
		w := parseInt(m[2])
		h := parseInt(m[3])

		// --- Puzzle 1 ---
		a := l * w
		b := w * h
		c := l * h
		n := min(a, b, c)
		total += 2*a + 2*b + 2*c + n

		// --- Puzzle 2 ---
		n = max(l, w, h)
		ribbon += 2*(l+w+h-n) + l*w*h
	}

	// --- Results ---
	fmt.Println("1. Total square feet of wrapping paper:", total)
	fmt.Println("2. Total feet of ribbon:", ribbon)
}
